src/core/Akka/Dispatch/MessageQueues/UnboundedPriorityMessageQueue.cs

```diff
@@ -14,9 +15,11 @@ namespace Akka.Dispatch.MessageQueues
     /// <summary> 
     /// Base class for a message queue that uses a priority generator for messages 
     /// </summary>
-    public class UnboundedPriorityMessageQueue : BlockingMessageQueue
+    public class UnboundedPriorityMessageQueue : BlockingMessageQueue, IUnboundedDequeBasedMessageQueueSemantics
```

Comment:
On the fence about this - it definitely adds deque semantics to the mailbox, but I'm now left wondering if we should be allowing stashing to work with the priority mailboxes in the first place.

If I think about it, the entire point of the prioritymailbox is to process messages in the order of the priority generator. FIFO no longer takes hold once you allow prioritization. So what good is stashing then? If I unstash a message I couldn't previously process, it either gets pushed back onto the queue in a totally different order (possibly not getting processed next) or if we stick with our normal double-ended queue semantics, it doesn't get processed in its normal priority order again.

I dunno, I'm leaning towards telling the users "don't use priority mailboxes with stashing." Either we just re-add messages we can't process back into the priority queue and re-prioritize them, which doesn't let us preserve message order (the re-added message will appear to be newer) or maybe we have to rewrite the prepend buffer to behave like a priority queue as well and process that first.